tests/ui/bool_to_int_with_if.rs

```diff
@@ -0,0 +1,66 @@
+// run-rustfix
+
+#![warn(clippy::bool_to_int_with_if)]
+#![allow(unused, dead_code, clippy::unnecessary_operation, clippy::no_effect)]
+
+fn main() {
+    let a = true;
+    let b = false;
+
+    let x = 1;
+    let y = 2;
+
+    // Should lint
+    // precedence
+    if a {
+        1
+    } else {
+        0
+    };
+    if !a {
+        1
+    } else {
+        0
+    };
+    if a || b {
+        1
+    } else {
+        0
+    };
+    if cond(a, b) {
+        1
+    } else {
+        0
+    };
+    if x + y < 4 {
+        1
+    } else {
+        0
+    };
+
+    // Shouldn't lint
+    if a {
+        3
+    } else {
+        0
+    };
+    if a {
+        side_effect();
+        1
+    } else {
+        0
+    };
```

Comment:
Thanks for suggesting test cases.
Turns out second case produces invalid code. I'll fix it and think more carefully about checking parent node.

Feels like suggesting wrapping it in else is better. And since I'll have to work on that case anyway, it shouldn't take to much extra time to implement.

Also I think suggesting inversion in cases like this would help with consistency.

```rust
if a {
    0
} else {
    1
}

// suggest
usize::from(!a)
```

I can feel myself getting carried away a little bit. But it's a lot fun, and imo it's better to release a good lint later than a half-baked sooner.

Reply:
> Turns out second case produces invalid code. I'll fix it and think more carefully about checking parent node.

You might want to take a look at [`is_else_clause`](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/fn.is_else_clause.html). That should do the check for you.

> I can feel myself getting carried away a little bit. But it's a lot fun, and imo it's better to release a good lint later than a half-baked sooner.

Getting carried away is a problem, I also have. The good thing is, that that is totally fine here. That's also a thing I enjoy about this and other open source projects. The focus is on the community and not on any deadline. So feel free to take your time. :upside_down_face: Alternatively, we can also get this merge ready, and you can improve the lint later :)